boards/metro_m0/examples/usb_serial.rs

```diff
@@ -181,75 +37,74 @@ fn main() -> ! {
     let mut red_led = pins.d13.into_open_drain_output(&mut pins.port);
     red_led.set_high().unwrap();
 
-    dbgprint!("make usb_bus");
+    //dbgprint!("make usb_bus");
 
-    let bus = unsafe {
-        USB_BUS = Some(hal::usb_bus(
-            peripherals.USB,
-            &mut clocks,
-            &mut peripherals.PM,
-            pins.usb_dm,
-            pins.usb_dp,
-            &mut pins.port,
+    let bus_allocator = unsafe {
+        USB_ALLOCATOR = Some(usb_allocator(
+                peripherals.USB,
+                &mut clocks,
+                &mut peripherals.PM,
+                pins.usb_dm,
+                pins.usb_dp,
+                &mut pins.port,
         ));
-        USB_BUS.as_ref().unwrap()
+        USB_ALLOCATOR.as_ref().unwrap()
     };
 
-    dbgprint!("make serial");
-    let serial = unsafe {
-        USB_SERIAL = Some(cdc_acm::SerialPort::new(bus));
-        USB_SERIAL.as_ref().unwrap()
+    unsafe {
+        USB_BUS = Some(
+            UsbDeviceBuilder::new(&bus_allocator, UsbVidPid(0x16c0, 0x27dd))
+                .manufacturer("Fake company")
+                .product("Serial port")
+                .serial_number("TEST")
+                .device_class(USB_CLASS_CDC)
+                .build(),
+            );
     };
 
-    dbgprint!("make dev");
+    //dbgprint!("make serial");
     unsafe {
-        USB_DEV = Some(
-            UsbDevice::new(bus, UsbVidPid(0x5824, 0x27dd))
-                .manufacturer("Wez Furlong")
-                .product("Serial port")
-                .serial_number("RUST")
-                .device_class(cdc_acm::USB_CLASS_CDC)
-                .build(&[serial]),
-        );
-    }
+        USB_SERIAL = Some(SerialPort::new(&bus_allocator));
+    };
 
     unsafe {
-        core.NVIC.set_priority(hal::Interrupt::USB, 0);
+        core.NVIC.set_priority(interrupt::USB, 0);
```

Comment:
I’m a little concerned about the dual case import continuing to work, is it guaranteed somewhere 🤔

Reply:
Yeah, I have no idea about this one. Another copy-paste 🤞

Reply:
:+1: lets switch to the capitalised `Interrupt` since that’s the type name in the PAC

Reply:
Sounds good. It looks like it is [aliased to the lower case for when rt is enabled](https://github.com/atsamd-rs/atsamd/blob/master/pac/atsamd21g18a/src/lib.rs#L169), but upper case does feel more correct here anyway.